base-files: enable get_mac_ascii() to handle redundant strings

DragonBluep opened the pull request:
The function get_mac_ascii() will fail when there are two or more same MAC address variable names in the mtd partition. Only retain the first variable to workaround this rare situation.

Tested-by: @huytran123. If you want a public Tested-by: tag on the commit message, please provide your name and email.

Please cherry-pick it to all stable branch.

DragonBluep committed:
base-files: enable get_mac_ascii() to handle redundant strings

The function get_mac_ascii() will fail when there are two or more
same MAC address variable names in the mtd partition. Only retain
the first variable to workaround this rare situation.

Fixes: https://github.com/openwrt/openwrt/issues/17236
Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/17262
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

neheb commented on package/base-files/files/lib/functions/system.sh:
```diff
@@ -66,7 +66,7 @@ get_mac_ascii() {
 	local mac_dirty
 
-	mac_dirty=$(strings "$part" | sed -n 's/^'"$key"'=//p')
+	mac_dirty=$(strings "$part" | sed -n 's/^'"$key"'=//p' | head -1)
```
hmmmm should be head -n 1 I think.

head -1 can be disabled for busybox IIRC.

ptpt52 replied:
another issue:
on some case, the data `Ffac_mac = XX:XX:XX:XX:XX:XX`
then the get_mac_ascii() get nothing

DragonBluep replied:
> another issue: on some case, the data `Ffac_mac = XX:XX:XX:XX:XX:XX` then the get_mac_ascii() get nothing

Fixed in https://github.com/openwrt/openwrt/pull/17262/commits/bc596b5f751c4ad66b28af24709ae7a03c7fe364.

DragonBluep committed:
base-files: strip space and tab characters from ASCII mac address

Spaces and tabs are widely used in variable definitions. We have to
remove them to ensure that get_mac_ascii() works properly.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/17262
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

1715173329 commented:
cc @hauke @robimarko would you have a look please?
